### PR TITLE
CRS-1849: latest calculation API

### PR DESCRIPTION
**Pull request by james-baker-digital-justice**

### CODE REVIEW

**Review by srowlands-moj — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/OffenderKeyDates.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  val topupSupervisionExpiryDate: LocalDate? = null,
+  val earlyRemovalSchemeEligibilityDate: LocalDate? = null,
+  val effectiveSentenceEndDate: LocalDate? = null,
+  val sentenceLength: String? = null,
+  val homeDetentionCurfewApprovedDate: LocalDate? = null,
+  val tariffDate: LocalDate? = null,
+  val tariffExpiredRemovalSchemeEligibilityDate: LocalDate? = null,
+  val approvedParoleDate: LocalDate? = null,
+  val releaseOnTemporaryLicenceDate: LocalDate? = null,
+  val judiciallyImposedSentenceLength: String? = null,
```
Comment (srowlands-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/OffenderKeyDates.kt:27]: @joelstobart-moj will be disappointed that you didn't shorten this to JISL

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/LatestCalculationService.kt [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      prisonerCalculation.earlyRemovalSchemeEligibilityDate?.let { ReleaseDate(it, ReleaseDateType.ERSED) },
+      prisonerCalculation.tariffExpiredRemovalSchemeEligibilityDate?.let { ReleaseDate(it, ReleaseDateType.TERSED) },
+      prisonerCalculation.dtoPostRecallReleaseDate?.let { ReleaseDate(it, ReleaseDateType.DPRRD) },
+    )
+    return LatestCalculation(
+      prisonerId,
+      calculatedAt,
+      location,
+      reason,
+      calculationSource,
```
[the remark this answers is not in the document]
Comment (james-baker-digital-justice) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/LatestCalculationService.kt:134]: I'll check with Scott, they are all listed in the ticket but I'm not sure which should be displayed. For release dates panel we return all in the API and the component filters the relevant ones.
